### PR TITLE
Infer whether user has set a password or not on init, drop the setting

### PR DESCRIPTION
**Pull request by juan-cortes**
### Type
Bug fix

### What was the original bug
If we set a password or removed an existing one and quickly refreshed or closed the application this password flag/change was persisted but the associated data was not encrypted/decrypted accordingly, giving an error on further read attempts. This pr drops the need for that flag (there's still a runtime application state level flag) and will determine whether the app is locked or not by attempting to read the data.

For a refresh of the app using cmd + r we had to introduce a nuke to the cache of the loaded data forcing a new fetch to ensure that this inference of the lock state wasn't broken. I consulted with @meriadec before implementing this since he wrote the original persistence layer back in the day.

### How to test
Since this change affects the way we detect if the application is currently locked or not, we need to cover all the scenarios that involve a password change or interaction with it:
- Onboarding on a fresh install we have the opportunity to set a password
- Setting up a password in the settings of the app
- Locking the app at any point and confirming it's locked on refresh and relaunch
- Changing the password / removing the password and making sure that the password is still correct.

In addition to the initial scope of the pr and following the request from @gre it also includes a drop in the write attempts to the app.json file when the app is locked. This was invalidating the counter values cache if I understood it correctly and it affected performance. I don't know how to test this other than looking at the file size of app.json before locking the app and seeing if it grows (it shouldn't while locked)

### CODE REVIEW

File: src/renderer/init.js [hunk 1]
```diff
@@ -81,10 +81,14 @@ async function init() {
 
   const isMainWindow = remote.getCurrentWindow().name === "MainWindow";
 
-  if (initialSettings.hasPassword) {
+  let accounts;
+  try {
+    accounts = await getKey("app", "accounts", []);
+  } catch (e) {
+    // NB Failing to retrieve the accounts means the data is encrypted.
```
Comment (gre) [inline src/renderer/init.js]: That's a bit weird approach to me. What if it's just filesystem failure?. Can we at least filter with a specific error (if there is a way to identify it)

**Review by gre — COMMENTED** [review]
Good overall but will needs some testing. I suggest we also include the work needed by LL-2173 to mutualize the QA work that will be needed for this

File: src/renderer/init.js (hunk 1 above)
Comment (juan-cortes) [reply src/renderer/init.js]: 🤔 you are right, needs a rethink

File: src/renderer/init.js [hunk 2]
```diff
@@ -83,11 +83,11 @@ async function init() {
 
   const isMainWindow = remote.getCurrentWindow().name === "MainWindow";
 
-  if (initialSettings.hasPassword) {
-    store.dispatch(lock());
-  } else {
-    const accounts = await getKey("app", "accounts", []);
+  const accounts = await getKey("app", "accounts", []);
```
Comment (juan-cortes) [inline src/renderer/init.js:86]: @gre Due to the changes from https://github.com/LedgerHQ/lld-v2/pull/188/files#diff-70f2e46b1f570eb03352ea45e36240e2R16-R20 we no longer need to try/catch to infer this locked state

**Review by gre — APPROVED** [review]

**Comment by juan-cortes** [comment]
Updated the pr description and added some qa notes @gre

**Comment by juan-cortes** [comment]
@Arnaud97234 this should be ready for you now

**Review by nabil-brn — CHANGES_REQUESTED** [review]
**✓** Onboarding on a fresh install we have the opportunity to set a password
**✓** Setting up a password in the settings of the app
**✓** Changing the password / removing the password and making sure that the password is still correct.

**✗** Locking the app at any point and confirming it's locked on refresh and relaunch
- cmd + R then typing a wrong password triggers this error:
```
Error occurred in handler for 'setEncryptionKey': Error
    at new CustomError (/Users/nbourenane/workspace/lld-v2/node_modules/@ledgerhq/errors/lib/helpers.js:27:18)
    at Object.setEncryptionKey (webpack-internal:///./src/main/db/index.js:172:11)
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async electron/js2c/browser_init.js:1093:30 {
  name: 'DBWrongPassword',
  message: 'DBWrongPassword',
  stack: 'Error\n' +
    '    at new CustomError (/Users/nbourenane/workspace/lld-v2/node_modules/@ledgerhq/errors/lib/helpers.js:27:18)\n' +
    '    at Object.setEncryptionKey (webpack-internal:///./src/main/db/index.js:172:11)\n' +
    '    at processTicksAndRejections (internal/process/task_queues.js:85:5)\n' +
    '    at async electron/js2c/browser_init.js:1093:30'
}
```


Additional tests:
**✗** app.json
- When password is enabled, this property is added: `"accounts":"v+JGHQlBz6uszCGHhxffWjpetyu8XtJHt4iElmLygvBe"`
- But this one remains unchanged: `"password":{"isEnabled":false}`

**✗** Auto-lock
Auto-lock is not working anymore, anything related?
Maybe if password.isEnabled = false it won't trigger the timer for autolock?

---
bda121d (HEAD -> hasPassword, ledger/hasPassword) Infer hasPassword, drop the setting, prevent locked write

**Comment by juan-cortes** [comment]
I completely forgot about the auto-lock (the irony), let me check these points and get back to you

**Comment by juan-cortes** [comment]
- Autolock was an issue with the `Idler` component not being ported correctly, the logic was broken so it never locked the app. I fixed it.
- The leftover setting was indeed forgotten but shouldn't have caused any issues, I see the value for `accounts` as something that fits if you didn't have any accounts since we are encrypting the text value of `[]` 
- I went over the onboarding to remove this and ensure that going back from the analytics step didn't break, it was breaking before I think.
- As for the error thrown, do you mean it's thrown crashing the app or in the console? I believe that's an error when it tries to decrypt with the wrong password and it's handled before it reaches the UI. If you mean it crashes the app, let me know 🤔

**Comment by nabil-brn** [comment]
✓ Autolock is back!
✓ The error thrown (not a crash) is still present but it'll be another ticket for this as discussed with @juan-cortes
✓ password.isEnabled removed from settings
```
% grep password ~/Library/Application\ Support/Electron/app.json
1
```
✓ Going back from the analytics step test case

I also retested everything listed before to ensure that no regression were introduced in the last 2 commits.

Edit: I've also tested a migration from an old app.json and the passlock was still enabled ✓.

✓ QA-ed

---
e967a83 (ledger/hasPassword) Fix idler, cleanup unused settings for password

**Review by nabil-brn — APPROVED** [review]

**Pull request merged**